### PR TITLE
Supporting VeBAL gauge voting

### PR DESCRIPTION
**Pull request by weitianjie2000**

### CODE REVIEW

File: contracts/TreasuryManager.sol [hunk 1]
```diff
@@ -149,6 +188,11 @@ contract TreasuryManager is
         emit NOTEPurchaseLimitUpdated(purchaseLimit);
     }
 
+    function setBALPurchaseLimit(uint256 purchaseLimit) external onlyOwner {
+        balPurchaseLimit = purchaseLimit;
```
Comment (jeffywu) [inline contracts/TreasuryManager.sol:192]: We should mark this as denominated in WETH.

File: tests/test_treasury_manager.py [hunk 2]
```diff
@@ -358,3 +358,35 @@ def test_invest_eth():
...
+    assert env.balLiquidityToken.balanceOf(env.treasuryManager.address) == 0
+    env.treasuryManager.addBalancerLiquidity(0, 1000e18, 0, {"from": testAccounts.testManager})
+    assert pytest.approx(env.balLiquidityToken.balanceOf(env.treasuryManager.address), abs=1000) == 414349726186765950603
+
+    # Delegate liquidity to VeBalDelegator
+    env.treasuryManager.delegateBalancerLiquidity(
+        env.balLiquidityToken.balanceOf(env.treasuryManager.address), 
+        {"from": testAccounts.testManager}
+    )
+    assert env.balLiquidityToken.balanceOf(env.treasuryManager.address) == 0
```
Comment (jeffywu) [inline tests/test_treasury_manager.py:380]: We should add some more tests to make sure we can lock and vote on the VeBalDelegator?

File: contracts/VeBalDelegator.sol [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+import "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+import "../interfaces/balancer/ILiquidityGauge.sol";
+import "../interfaces/balancer/ILiquidityGaugeController.sol";
+import "../interfaces/balancer/IVeToken.sol";
+import "../interfaces/notional/IStakedNote.sol";
+
+/// @title 80-BAL-20-WETH BPT PCV Deposit
+/// @author Fei Protocol
+contract VeBalDelegator is
+    VoteEscrowTokenManager,
```
Comment (jeffywu) [inline contracts/VeBalDelegator.sol:16]: Was there a snapshot delegator in the original Fei code? Balancer does a lot of voting on snapshot so we might want to make sure we can delegate for that:

https://docs.snapshot.org/guides/delegation

**Review by jeffywu — CHANGES_REQUESTED** [review]

File: contracts/VeBalDelegator.sol (hunk 3 above)
Comment (weitianjie2000) [reply contracts/VeBalDelegator.sol:18]: yeah, I added the snapshot delegator back.

File: tests/test_treasury_manager.py (hunk 2 above)
Comment (weitianjie2000) [reply tests/test_treasury_manager.py:380]: done

File: contracts/TreasuryManager.sol (hunk 1 above)
Comment (weitianjie2000) [reply contracts/TreasuryManager.sol:194]: one

**Pull request closed without merging**